Disabling schedulers causes build errors

Whilst attempting to locate a link error for missing libatomic symbols, I disabled some schedulers as I suspected on of them is the culprit.
This leads to errors of the kind
```
/home/biddisco/src/hpx/src/runtime/threads/threadmanager.cpp: In member function ‘void hpx::threads::threadmanager::create_pools()’:
/home/biddisco/src/hpx/src/runtime/threads/threadmanager.cpp:538:31: error: ‘command_line_error’ is not a member of ‘hpx::threads::detail’
                 throw detail::command_line_error(
                               ^~~~~~~~~~~~~~~~~~
/home/biddisco/src/hpx/src/runtime/threads/threadmanager.cpp:538:31: note: suggested alternative:
```
and
```
/home/biddisco/src/hpx/src/runtime/threads/threadmanager.cpp:689:21: error: ‘periodic_priority_queue_scheduler’ in namespace ‘hpx::threads::policies’ does not name a template type
                     periodic_priority_queue_scheduler<>
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/biddisco/src/hpx/src/runtime/threads/threadmanager.cpp:691:17: error: ‘local_sched_type’ has not been declared
                 local_sched_type::init_parameter_type init(num_threads_in_pool,
```

--- a/src/runtime/threads/threadmanager.cpp
+++ b/src/runtime/threads/threadmanager.cpp
@@ -10,6 +10,7 @@
 #include <hpx/config.hpp>
 #include <hpx/compat/mutex.hpp>
 #include <hpx/exception.hpp>
+#include <hpx/error_code.hpp>
 #include <hpx/performance_counters/counter_creators.hpp>
 #include <hpx/performance_counters/counters.hpp>
 #include <hpx/performance_counters/manage_counter_type.hpp>
@@ -419,8 +420,7 @@ namespace hpx { namespace threads
 
 #else
                 throw detail::command_line_error(
-                    "Command line option "
-                    "--hpx:queuing=local "
+                    "Command line option --hpx:queuing=local "
                     "is not configured in this build. Please rebuild with "
                     "'cmake -DHPX_WITH_THREAD_SCHEDULERS=local'.");
 #endif
@@ -536,8 +536,7 @@ namespace hpx { namespace threads
 
 #else
                 throw detail::command_line_error(
-                    "Command line option "
-                    "--hpx:queuing=static "
+                    "Command line option --hpx:queuing=static "
                     "is not configured in this build. Please rebuild with "
                     "'cmake -DHPX_WITH_THREAD_SCHEDULERS=static'.");
 #endif
@@ -583,8 +582,7 @@ namespace hpx { namespace threads
 
 #else
                 throw detail::command_line_error(
-                    "Command line option "
-                    "--hpx:queuing=static-priority "
+                    "Command line option --hpx:queuing=static-priority "
                     "is not configured in this build. Please rebuild with "
                     "'cmake -DHPX_WITH_THREAD_SCHEDULERS=static-priority'.");
 #endif
@@ -625,8 +623,7 @@ namespace hpx { namespace threads
                 pools_.push_back(std::move(pool));
 #else
                 throw detail::command_line_error(
-                    "Command line option "
-                    "--hpx:queuing=abp-priority "
+                    "Command line option --hpx:queuing=abp-priority "
                     "is not configured in this build. Please rebuild with "
                     "'cmake -DHPX_WITH_THREAD_SCHEDULERS=abp-priority'.");
 #endif
@@ -666,8 +663,7 @@ namespace hpx { namespace threads
                 pools_.push_back(std::move(pool));
 #else
                 throw detail::command_line_error(
-                    "Command line option "
-                    "--hpx:queuing=hierarchy "
+                    "Command line option --hpx:queuing=hierarchy "
                     "is not configured in this build. Please rebuild with "
                     "'cmake -DHPX_WITH_THREAD_SCHEDULERS=hierarchy'.");
 #endif
@@ -676,6 +672,7 @@ namespace hpx { namespace threads
 
             case resource::periodic_priority:
             {
+#if defined(HPX_HAVE_PERIODIC_PRIORITY_SCHEDULER)
                 // set parameters for scheduler and pool instantiation and
                 // perform compatibility checks
                 hpx::detail::ensure_hierarchy_arity_compatibility(cfg_.vm_);
@@ -705,7 +702,12 @@ namespace hpx { namespace threads
                             policies::delay_exit),
                         thread_offset));
                 pools_.push_back(std::move(pool));
-
+#else
+                throw detail::command_line_error(
+                    "Command line option --hpx:queuing=periodic-priority "
+                    "is not configured in this build. Please rebuild with "
+                    "'cmake -DHPX_WITH_THREAD_SCHEDULERS=periodic-priority'.");
+#endif
                 break;
             }
 
@@ -742,8 +744,7 @@ namespace hpx { namespace threads
 
 #else
                 throw hpx::detail::command_line_error(
-                    "Command line option "
-                    "--hpx:queuing=throttle "
+                    "Command line option --hpx:queuing=throttle "
                     "is not configured in this build. Please rebuild with "
                     "'cmake -DHPX_WITH_THREAD_SCHEDULERS=throttle "
                     "-DHPX_WITH_APEX'.");

